compiler/stmt.py

```diff
@@ -41,6 +63,7 @@ def __init__(self, block_):
     self.block = block_
     self.writer = util.Writer()
     self.expr_visitor = expr_visitor.ExprVisitor(self.block, self.writer)
+    self.parser_flags = 0
```

Comment:
I think we can get away with a single pass. According to [the 2.x docs](https://docs.python.org/2/reference/simple_stmts.html#future):

> A future statement must appear near the top of the module. The only lines that can appear before a future statement are:
> 
> * the module docstring (if any),
> * comments,
> * blank lines, and
> * other future statements.

None of which any future feature would alter the parsing logic of. So I think we can safely proceed with any flags after they are encountered.